### PR TITLE
Fixed negative netty sleep values.

### PR DESCRIPTION
**Pull request by brndnmtthws**
For some context, it seems that retries.get() occasionally returns negative values.

### CODE REVIEW

File: storm-netty/src/jvm/backtype/storm/messaging/netty/Client.java [hunk 1]
```diff
@@ -93,9 +93,11 @@ void reconnect() {
      */
     private int getSleepTimeMs()
     {
-        int sleepMs = base_sleep_ms * Math.max(1, random.nextInt(1 << retries.get()));
+        int sleepMs = base_sleep_ms * Math.max(1, Math.min(1, random.nextInt(1 << retries.get())));
```
Comment (anfeng) [inline storm-netty/src/jvm/backtype/storm/messaging/netty/Client.java]: Why do we have min() within max()?

Comment (brndnmtthws) [inline storm-netty/src/jvm/backtype/storm/messaging/netty/Client.java]: Woops, I effed the patch. Good catch.  I just updated the patch.

File: storm-netty/src/jvm/backtype/storm/messaging/netty/Client.java [hunk 2]
```diff
@@ -93,9 +93,12 @@ void reconnect() {
      */
     private int getSleepTimeMs()
     {
-        int sleepMs = base_sleep_ms * Math.max(1, random.nextInt(1 << retries.get()));
+        int backoff = 1 << Max(1, retries.get());
+        int sleepMs = base_sleep_ms * Math.max(1, random.nextInt(backoff));
```
Comment (anfeng) [inline storm-netty/src/jvm/backtype/storm/messaging/netty/Client.java:97]: why change Line 96 at all?

Comment (brndnmtthws) [inline storm-netty/src/jvm/backtype/storm/messaging/netty/Client.java:97]: I don't understand your comment.  I wanted to make the code more readable.

**Comment by anfeng** [comment]
+1

**Comment by ptgoetz** [comment]
+1

**Comment by xumingming** [comment]
+1

**Comment by ptgoetz** [comment]
Tagged for 0.9.0

**Pull request merged**